sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -184,15 +189,65 @@ abstract class JdbcDialect extends Serializable {
   /**
    * Rename an existing table.
    *
-   * TODO (SPARK-32382): Override this method in the dialects that don't support such syntax.
-   *
    * @param oldTable The existing table.
    * @param newTable New name of the table.
    * @return The SQL statement to use for renaming the table.
    */
   def renameTable(oldTable: String, newTable: String): String = {
     s"ALTER TABLE $oldTable RENAME TO $newTable"
   }
+
+  /**
+   * Alter an existing table.
+   *
+   * @param tableName The name of the table to be altered.
+   * @param changes Changes to apply to the table.
+   * @return The SQL statements to use for altering the table.
+   */
+  def alterTable(tableName: String, changes: Seq[TableChange]): Array[String] = {
+    val updateClause = ArrayBuilder.make[String]
+    for (change <- changes) {
+      change match {
+        case add: AddColumn =>
+          add.fieldNames match {
+            case Array(name) =>
+              val dataType = JdbcUtils.getJdbcType(add.dataType(), this).databaseTypeDefinition
+              updateClause += s"ALTER TABLE $tableName ADD COLUMN $name $dataType"
+          }
```

Comment:
I would implement this (and below) in more reliable way:
 ```suggestion
          for (name <- add.fieldNames) {
            val dataType = JdbcUtils.getJdbcType(add.dataType(), this).databaseTypeDefinition
            updateClause += s"ALTER TABLE $tableName ADD COLUMN $name $dataType"
          }
```
I am not sure that we can guarantee that the array always contain 1 element.

Reply:
Seems array always contains 1 element? 
because ```sql("ALTER TABLE h2.test.alt_table ADD COLUMNS (C1 INTEGER, C2 STRING)")``` will generate two AddColumn (with Array(C1) and Array(C2) in fieldNames respectively), and we don't support ```sql("ALTER TABLE h2.test.alt_table ADD COLUMNS (C1  C2 STRING)")```

Reply:
Probably, you are right.